MekHQ/src/mekhq/gui/dialog/nagDialogs/UntreatedPersonnelNagDialog.java

```diff
@@ -30,12 +30,16 @@
 public class UntreatedPersonnelNagDialog extends AbstractMHQNagDialog {
     private static boolean isUntreatedInjury (Campaign campaign) {
         for (Person p : campaign.getActivePersonnel()) {
-            for (Injury i : p.getInjuries()) {
-                if (!i.isPermanent()) {
-                    if (p.getDoctorId() == null) {
-                        return true;
+            if (campaign.getCampaignOptions().isUseAdvancedMedical()) {
+                for (Injury i : p.getInjuries()) {
+                    if (!i.isPermanent()) {
+                        return p.getDoctorId() == null;
                     }
                 }
+            } else if (!campaign.getCampaignOptions().isUseAdvancedMedical()) {
```

Comment:
No need for the if here. Just do `else {`.

Reply:
So you'd think that, and I agree, that's how it _should_ work. However, when I was testing it was triggering the dialog twice whenever the 'else if' wasn't in place. It's beyond me as to why.

Reply:
Huh, that shouldn't happen.

Reply:
I am not aware of how advanced medical works, but do you even need  both checks. Isn't it also hit dependent? i.e. if someone has zero hits, they have no non-permeanant wounds.

Reply:
Found a better solution to all of this: `person.needsFixing()`. That will check both basic hits and advanced medical in one check and simplify your approach.

```java
for (Person p : campaign.getActivePersonnel()) {
    if(p.needsFixing() && p.getDoctorId() == null) {
       return true;
   }
}
```

that will cover both cases.

Reply:
Done. I went ahead and used your suggested code and it works like a charm.

I opted not to check for permanent injuries, as personnel with permanent injuries still need to be assigned to a doctor, so I doubt anyone will complain.